### PR TITLE
Add activeStates metrics per state

### PR DESCRIPTION
**Pull request by fabriziodemaria**

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java [hunk 1]
```diff
@@ -453,8 +453,11 @@ private void setupMetrics(
       }
     };
 
+    Arrays.stream(RunState.State.values()).forEach(state -> stats.registerActiveStates(
```
Comment (bergman) [inline styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: doesn't `values()` have a `forEach`?

`RunState.State.values().foreach(state -> ...`

File: styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java [hunk 2]
```diff
@@ -435,7 +436,6 @@ private void setupMetrics(
       Stats stats) {
 
     final Gauge<Long> queuedEventsCount = stateManager::getQueuedEventsCount;
-    final Gauge<Long> activeStatesCount = stateManager::getActiveStatesCount;
```
Comment (bergman) [inline styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: I would keep the old one for now, we can remove it later

File: styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java (hunk 1 above)
Comment (fabriziodemaria) [inline styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: That doesn't seem to work with enum...

File: styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java (hunk 2 above)
Comment (fabriziodemaria) [inline styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: The same graph achieved by the old one can still be constructed from the new one I believe

Comment (rouzwawi) [inline styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: Agree

File: styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java (hunk 1 above)
Comment (bergman) [inline styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java]: ah, didn't notice it was an array

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java [hunk 3]
```diff
@@ -122,8 +125,13 @@ public void registerQueuedEvents(Gauge<Long> queuedEventsCount) {
   }
 
   @Override
-  public void registerActiveStates(Gauge<Long> activeStatesCount) {
-    registry.register(ACTIVE_STATES, activeStatesCount);
+  public void registerActiveStates(RunState.State state, Gauge<Long> activeStatesCount) {
+    registry.register(ACTIVE_STATES_PER_RUNSTATE, activeStatesCount);
```
Comment (rouzwawi) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: This line will register different gauges for the same metricId. That will most likely lead to confusing results.

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java [hunk 4]
```diff
@@ -86,6 +87,7 @@
   private final ConcurrentMap<String, Histogram> dockerOperationHistograms;
   private final ConcurrentMap<String, Meter> dockerOperationMeters;
   private final ConcurrentHashMap<WorkflowId, Gauge> activeStatesPerWorkflowGauges;
+  private final ConcurrentHashMap<RunState.State, Gauge> activeStatesPerRunStateGauges;
```
Comment (rouzwawi) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: This map is not needed as the set of states is always static and we set up gauges on startup

**Comment by rouzwawi** [comment]
LGTM

**Pull request merged**